assets/js/frontend/single-product.js

```diff
@@ -154,68 +174,81 @@ jQuery( function( $ ) {
 			// But only zoom if the img is larger than its container.
 			if ( enable_zoom ) {
 				zoom_target.trigger( 'zoom.destroy' );
-				zoom_target.zoom({
+				zoom_target.zoom( {
 					touch: false
-				});
+				} );
 			}
-		},
+		};
+
+		this.reset_slide_position = function() {
+
+			if ( ! gallery.flexslider_enabled ) {
+				return;
+			}
+
+			$el.flexslider( 0 );
+		};
 
 		/**
 		 * Get product gallery image items.
 		 */
-		get_gallery_items: function() {
-			var $slides = $( '.woocommerce-product-gallery__wrapper' ).children(),
-				items   = [],
-				index   = $slides.filter( '.' + 'flex-active-slide' ).index();
+		this.get_gallery_items = function() {
+
+			var $slides = this.$images,
+				items   = [];
 
 			if ( $slides.length > 0 ) {
 				$slides.each( function( i, el ) {
 					var img = $( el ).find( 'img' ),
-						large_image_src = img.attr( 'data-large-image' ),
-						large_image_w   = img.attr( 'data-large-image-width' ),
-						large_image_h   = img.attr( 'data-large-image-height' ),
+						large_image_src = img.attr( 'data-large_image' ),
+						large_image_w   = img.attr( 'data-large_image_width' ),
+						large_image_h   = img.attr( 'data-large_image_height' ),
 						item            = {
 							src: large_image_src,
 							w:   large_image_w,
 							h:   large_image_h,
 							title: img.attr( 'title' )
 						};
 					items.push( item );
-				});
+				} );
 			}
 
-			return {
-				index: index,
-				items: items
-			};
-		},
+			return items;
+		};
 
 		/**
 		 * Init PhotoSwipe.
 		 */
-		init_photoswipe: function() {
-			if ( wc_single_product_params.zoom_enabled ) {
-				$( '.woocommerce-product-gallery--with-images' ).prepend( '<a href="#" class="woocommerce-product-gallery__trigger">🔍</a>' );
-				$( document ).on( 'click', '.woocommerce-product-gallery__trigger', this.trigger_photoswipe );
+		this.init_photoswipe = function() {
+
+			if ( ! this.photoswipe_enabled ) {
+				return;
+			}
+
+			if ( this.zoom_enabled && this.$images.length > 0 ) {
+				$el.prepend( '<a href="#" class="woocommerce-product-gallery__trigger">🔍</a>' );
+				$el.on( 'click', '.woocommerce-product-gallery__trigger', { gallery: this }, this.trigger_photoswipe );
 			}
-			$( document ).on( 'click', '.woocommerce-product-gallery__image a', this.trigger_photoswipe );
-		},
+
+			$el.on( 'click', '.woocommerce-product-gallery__image a', { gallery: this }, this.trigger_photoswipe );
+		};
 
 		/**
 		 * Initialise photoswipe.
 		 */
-		trigger_photoswipe: function( e ) {
+		this.trigger_photoswipe = function( e ) {
+
 			e.preventDefault();
 
 			var pswpElement = $( '.pswp' )[0],
-				items  = wc_product_gallery.get_gallery_items(),
+				items  = e.data.gallery.get_gallery_items(),
 				target = $( e.target ),
 				clicked;
 
 			if ( ! target.is( '.woocommerce-product-gallery__trigger' ) ) {
-				clicked = e.target.closest( 'figure' );
+				clicked = e.target.closest( '.woocommerce-product-gallery__image' );
```

Comment:
Also isn't this `figure` changed to a `div` in 53b7844412425a2d747cf49e4955c5c865210dc1 ?